lib/ocpp/v201/charge_point.cpp

```diff
@@ -3296,6 +3393,132 @@ void ChargePoint::handle_heartbeat_response(CallResult<HeartbeatResponse> call)
     }
 }
 
+void ChargePoint::handle_reserve_now_request(Call<ReserveNowRequest> call) {
+    ReserveNowResponse response;
+    response.status = ReserveNowStatusEnum::Rejected;
+    if (!this->callbacks.reserve_now_callback.has_value() ||
+        !this->device_model->get_optional_value<bool>(ControllerComponentVariables::ReservationCtrlrAvailable)
+             .value_or(false) ||
+        !this->device_model->get_optional_value<bool>(ControllerComponentVariables::ReservationCtrlrEnabled)
+             .value_or(false)) {
+        // Reservation not available / implemented, return 'Rejected'.
+        // H01.FR.01
+        EVLOG_info << "Receiving a reservation request, but reservation is not implemented.";
```

Comment:
if `ReservationCtrlrEnabled` is false it might be implemented, just not enabled.

Also, for most of these it would be nice if we can add the error to the call result as well for the CSMS to see what is actually happening.